lab_4_summarization_textrank/README.md

```diff
@@ -0,0 +1,758 @@
+
+**Важно**: в ситуации, когда у текстов одинаковые значения метрики похожести, необходимо сортировать значения по пути до текста в **обратном** (по убыванию) лексикографическом порядке.
+
+Интерфейс:
+```py
+def _find_texts_close_to_keywords(self,
+                                  keywords: tuple[str, ...],
+                                  n_texts: int) -> tuple[str, ...]:
+    pass
+```
+
+**Важно №2**: в ситуации, когда у всех текстов значение метрики похожести на ключевые слова равняется 0, необходимо поднять 
+_пользовательское исключение_ с сообщением `Не было найдено текстов, подходящих под запрос. Попробуйте другой запрос.`.
+
+Создайте пользовательское исключение с именем `NoRelevantTexts`. Что такое пользовательские исключения и как их создавать, Вы можете
+найти [здесь](https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions).
+
+#### Шаг 9.3 Внешний интерфейс помощника
+
+Теперь, когда Вы знаете, как находить наиболее похожие на запрос тексты и имеете доступ к их кратким содержаниям, 
+пришла пора последнего шага: создания внешнего интерфейса помощника, который будет использовать реализованное выше.
+
+Конечная цель, обозначенная ранее: создание помощника, который может отвечать на запрос пользователя определённым количеством кратких содержаний.
+Эти краткие содержания должны быть релевантны запросу.
+
+Для того чтобы помощник мог взаимодействовать с пользователем, Вам нужно реализовать метод `reply`.
+
+Метод принимает на вход запрос и количество кратких содержаний, которые будут в ответе (по умолчанию - три). 
+Если в базе данных количество текстов меньше, чем количество необходимых кратких содержаний, поднимается ошибка `ValueError`. 
+
+Метод возвращает ответ со следующим форматированием:
+```python
+Ответ:
+{SUMMARY_1}
+
+{SUMMARY_2}
+...
+
+{SUMMARY_N}
+```
+
+Краткие содержания, соединённые двумя символами новой строки.
+
+**Важно**: количество предложений в кратких содержаниях должно равняться 5.
+
+Интерфейс:
+```py
+def reply(self, query: str, n_summaries: int = 3) -> str:
+    pass
+```
+
+**Важно №2**: в ситуации, когда запрос пользователя имеет некорректный тип или является пустой строкой, необходимо поднять 
+_пользовательское исключение_ с сообщением `Некорректный запрос. Введите строку.`.
+
+Создайте пользовательское исключение с именем `IncorrectQuery`. Что такое пользовательские исключения и как их создавать, Вы можете
+найти [здесь](https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions).
+
+### Шаг 10. Продемонстрировать работу всезнающего помощника. Выполнение Шагов 1-10 соответствует 10 баллам
+
+Продемонстрируйте взаимодействие с помощником: обучите его с помощью текстов, хранящихся в папке [texts](./assets/texts), 
+задайте несколько вопросов и оцените, насколько полезными оказались ответы.
+
+**Важно**: переменная `paths_to_texts` хранит список **путей** до текстов, используйте её при инициализации помощника.
```

Comment:
is not it too late to know about it?

Reply:
It is also stated in the very beginning in the listing of the predefined variables.